Model/DutyCalculator.php

```diff
@@ -261,36 +280,35 @@ public function fillOutQuoteAndSessionUsingFeeReturned($duty, $address, $apply,
         $quote = $this->checkoutSession->getQuote();
         $duty_adjusted = $this->priceCurrency->round($duty); //copied over pre-existing code
         //but are we supposed to round always?
-        //should not that be based on corresponding admin setting?
+        //should not that be based on corresponding admin setting? or what is coming back from REACH API
         $this->checkoutSession->setReachDuty($duty_adjusted);
 
+        $baseDuty = $this->getBaseDuty($duty_adjusted);
+
+        $this->checkoutSession->setBaseReachDuty($baseDuty);
+
+        $this->checkoutSession->setDhlQuoteId($response['quoteId']);
+        //$this->checkoutSession->setReachDuty($duty_adjusted);
+        $this->checkoutSession->setDhlBreakdown(json_encode($response['feeTotals']));
+
         if ($apply || !$this->getIsOptional($address->getCountryId())) {
             //checkbox selection was duty should be applied
             //or applying duty is a must for that country
-            $baseCurrency = $this->storeManager->getStore()->getBaseCurrency();
-            $rate = $baseCurrency->getRate($baseCurrency->getCode());
-            $baseDuty = $duty_adjusted / $rate;
             $quote->setBaseReachDuty($baseDuty);
-            $this->checkoutSession->setBaseReachDuty($baseDuty);
             $quote->setReachDuty($duty_adjusted);
-            $this->checkoutSession->setReachDuty($duty_adjusted);
```

Comment:
with the removal of these session calls only a part of the session is being updated in this function that is named: `fillOutQuoteAndSessionUsingFeeReturned` could you help explain why these lines were removed? and potentially rename the function to better reflect what is actually happening here?

Let me know if I'm just wildly missing the point haha thank you!

Reply:
Actually the statements are taken out and placed outside that conditional block not entirely removed. This is because those are needed for both (if and else) cases if we want to use session as temporary memory and copied back to Quote model to save those value in the database (when the `apply D&T`  is pressed/forced without giving a the user a choice). If `apply d&t` checkbox is deselected then the copy from session to quote model does not happen though the D&T data is available in session if needed. The function is named as `fillOutQuoteAndSessionUsingFeeReturned` as this function fills out quote and session data structures with fees returned from DHL API call at some point. Thanks.

Reply:
thanks for the explanation! No this is all good, I needed to look at the function overall.